app.rb

```diff
@@ -0,0 +1,41 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age:, name: 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    is_of_age? || @parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
+end
+
+class Student < Person
+  def initialize(age:, classroom:, name: 'Unknown', parent_permission: true)
+    super(age: age, name: name, parent_permission: parent_permission)
+    @classroom = classroom
+  end
+
+  def play_hooky
+    '¯(ツ)/¯'
+  end
+end
+
+class Teacher < Person
+  def initialize(age:, specialization:, name: 'Unknown', parent_permission: true)
+    super(age: age, name: name, parent_permission: parent_permission)
+    @specialization = specialization
+  end
+
+  def can_use_services?
+    true
+  end
+end
```

Comment:
- [ ] Based on the requirements, there should be an individual .rb file for each class (Person, Student, and Teacher). Please create separate files for each of the classes above 👍🏿